### ISSUE TITLE
[Notifications] Chat notification handler is not working for me

### ISSUE THREAD

**Issue by rupinderjeet** [GetStream/stream-chat-android#1487]
I was already using Firebase Push Notifications before I added stream-chat in my project. I think stream-chat's push notifications are being treated in the same way as if they were from our backend server?

I have added a `ChatNotificationHandler` to my `ChatClient` instance. When I tap on received notification, no activity opens, notification gets dismissed/cancelled. In `NotificationConfig`, I added a drawable to be used as `smallIcon`, received notifications don't have it. So, either `ChatNotificationHandler` is not working, or I might be doing something wrong.

If I add following `intent-filter` to one of the activities in AndroidManifest, notifications do start that Activity. But, then, the intent returned in `getNewMessageIntent()` of `ChatNotificationHandler` is not used either. If I remove notification handler from `ChatClient`, this works without problem.

```
<intent-filter>
	<action android:name="OPEN_ACTIVITY_1" />
	<category android:name="android.intent.category.DEFAULT" />
</intent-filter>
```
i.e. it opens the activity without the handler as well.

When I try to retrieve `extras` from received `intent`, this is what I get:

```
[google.delivered_priority] = 'high'
[google.sent_time] = '1614584301807'
[google.ttl] = '2419200'
[google.original_priority] = 'high'
[sender] = '4389707'
[from] = '683576655095'
[google.message_id] = '0:1614584301819296%2e44d9f22e44d9f2'
[channel] = '{"id":"!members-kf9qL75JPYtFJ4618vb41CfC0qvXPV2OCaqArEPrU0Y","type":"messaging"}'
[message] = '4389707-8570bcf1-1864-4800-a20d-b3b8667de132'
[collapse_key] = '[redacted]'
```

Previously, I had a discussion on this in #528 

I want to ask if this is how I have to handle notifications, or am I doing something wrong?

**Comment by rupinderjeet** [comment]
I have a chat notification handler like this:

```
class AppChatNotificationHandler(
	context: Context,
	notificationConfig: NotificationConfig,
) : ChatNotificationHandler(context, notificationConfig) {

	override fun getNewMessageIntent(
		messageId: String,
		channelType: String,
		channelId: String,
	): Intent {

                 // TODO: add global constant keys
		return Intent(context, ChatStartupActivity::class.java).apply {
			putExtra("extra_channel_id", channelId)
			putExtra("extra_channel_type", channelType)
			putExtra("extra_message_id", messageId)
		}
	}
}
```

And, I assign it to `ChatClient` like this:

```
val notificationHandler = AppChatNotificationHandler(
	applicationContext,
	notificationConfig
)

val logLevel = if (BuildConfig.DEBUG) ChatLogLevel.ALL else ChatLogLevel.NOTHING

val client = ChatClient.Builder(API_KEY, applicationContext)
	.loggerHandler(AppChatSdkLogger)
	.logLevel(logLevel)
	.notifications(notificationHandler)
	.build()
```

**Comment by samiuelson** [comment]
Hey @rupinderjeet What SDK version are you using?

**Comment by rupinderjeet** [comment]
Hi @samiuelson, I have tested with both 4.5.5 and 4.6.0. Same results. But, I am working primarily with 4.5.5 right now.

**Comment by samiuelson** [comment]
What's the exact problem, does `Intent.getStringExtra()` return `null` for "extra_channel_id", "extra_channel_type", and "extra_message_id"?

**Comment by samiuelson** [comment]
Also, is there a chance you have registered your own custom `FirebaseMessagingService` in the app?

**Comment by rupinderjeet** [comment]
Yes. Also, `smallIcon` for notification is not from the defined chat notification-config.

I created a notification-channel `(Android)` for our backend's notifications where I set a different `smallIcon`
```
NotificationCompat.Builder(
    context, 
    AppNotificationChannels.DEFAULT_CHANNEL_ID
).setSmallIcon(R.drawable.ic_app_notification)
```

Stream's notifications are using this notification channel's `smallIcon`. So, I think config in `ChatNotificationHandler` is not being set somehow. I am testing by overriding other methods of handler to see if they are being called.

**Comment by rupinderjeet** [comment]
I have this declaration in my `app/` module

```
<service
	android:name=".integrations.firebase.PushNotificationsService"
	android:exported="false">
	<intent-filter>
		<action android:name="com.google.firebase.MESSAGING_EVENT" />
	</intent-filter>
</service>
```

And, I, also, had one from Stream docs, (I think intent-filters are conflicting declarations)

```
<service
	android:name="io.getstream.chat.android.livedata.service.sync.OfflineSyncFirebaseMessagingService"
	android:exported="false">
	<intent-filter>
		<action android:name="com.google.firebase.MESSAGING_EVENT" />
	</intent-filter>
</service>
```

I removed both of these from AndroidManifest, result is same. Keeping one of either is not helping either.

**Comment by samiuelson** [comment]
In genereal, there's no way of having multiple instances of `FirebaseMessagingService` working in one app. If you need to support notifications from a source other than Steam service, here's the recommended solution using `PushMessageSyncHandler` in your own `FirebaseMessagingService`: https://getstream.io/chat/docs/android/push_android/?language=kotlin#handling-notifications-from-multiple-backend-services

Could you also verify your Firebase settings at https://dashboard.getstream.io/? The template should not be populated, only the data template should have key-value pairs defined like this:
<img width="300" alt="Screenshot 2021-03-01 at 12 30 51" src="https://user-images.githubusercontent.com/4527432/109491999-97747780-7a8a-11eb-91f9-a2dc1872e268.png">
<img width="300" alt="Screenshot 2021-03-01 at 12 30 56" src="https://user-images.githubusercontent.com/4527432/109492008-9a6f6800-7a8a-11eb-90af-5c1f288f54f8.png">

**Comment by rupinderjeet** [comment]
Hello @samiuelson, can you give me a quick walkthrough on how to reach this dialog/setting in Stream Dashboard? I am unable to find it.

**Comment by samiuelson** [comment]
1. Open your app on Stream dashboard, choose Chat > Overview from the top-left menu. 
<img width="300" alt="Screenshot 2021-03-01 at 12 58 50" src="https://user-images.githubusercontent.com/4527432/109494220-e4a61880-7a8d-11eb-865a-f04d153fcefd.png">
2. Scroll down to the  Push Notifications section and click Edit Settings (next to Android&Firebase)
<img width="800" alt="Screenshot 2021-03-01 at 12 58 32" src="https://user-images.githubusercontent.com/4527432/109494233-e7a10900-7a8d-11eb-9994-3c382f614e91.png">

**Comment by rupinderjeet** [comment]
In `4.6.0`, everything is working for me after I changed templates as you said and added the handler in my `FirebaseMessagingService`. 

Also, I think `PushMessageSyncHandler` is only available in `4.6.0`

I am using `4.5.5`. Do you think there is something I can do for this version?

Thank you for your help.

**Comment by samiuelson** [comment]
Unfortunately, there's no easy way to handle push messages from multiple providers in pre-4.6.0 versions. I'd also recommend upgrading to 4.6.0 version because it brings stability improvements. 

If you really need to use older versions what you can do is to integrate the message and token handling logic from `OfflineSyncFirebaseMessagingService.kt` (from version 4.5.5) into your own `FirebaseMessagingService` subclass.
